### PR TITLE
Upgrade NeoFS SDK Go with another approach of container sessions

### PR DESCRIPTION
**Pull request by cthulhu-rider**
After recent changes in NeoFS SDK Go library session tokens aren't
embedded into `container.Container` and `eacl.Table` structures.
Instead, the operations of storing given values in NeoFS are
parameterized by elements of the corresponding type.

Add dedicated session parameters to operations of bucket and eACL
setting.

Signed-off-by: Leonard Lyubich <leonard@nspcc.ru>

* based on and blocked by https://github.com/nspcc-dev/neofs-sdk-go/pull/276

### CODE REVIEW

**Comment by KirillovDenis** [comment]
Can we unify container operations? I mean move this:
https://github.com/nspcc-dev/neofs-s3-gw/blob/8a1fc8ae3f99cff5dc060d969a384e94e109ec1f/api/layer/container.go#L189-L193
to `DeleteBucketHandler`. 

Also add check for `nil` here:
https://github.com/nspcc-dev/neofs-s3-gw/blob/8a1fc8ae3f99cff5dc060d969a384e94e109ec1f/internal/neofs/neofs.go#L203

**Comment by cthulhu-rider** [comment]
@KirillovDenis I'm a little puzzled: is it related to current PR's changes?

**Comment by KirillovDenis** [comment]
Actually not. Is related to container session tokens in general

**Review by KirillovDenis — APPROVED** [review]

**Comment by cthulhu-rider** [comment]
@KirillovDenis oh I think I've finally got your point, and agree. See 36a743c0fcbf97d9f12d31a4690c4dd2729304a2, eb9b5177a5774982f3809949d435ab228be941dd

File: api/layer/container.go [hunk 1]
```diff
@@ -180,12 +180,6 @@ func (n *layer) GetContainerEACL(ctx context.Context, idCnr cid.ID) (*eacl.Table
...
 }
 
-func (n *layer) deleteContainer(ctx context.Context, idCnr cid.ID) error {
-	var sessionToken *session.Container
-	boxData, err := GetBoxData(ctx)
-	if err == nil {
-		sessionToken = boxData.Gate.SessionTokenForDelete()
-	}
-
+func (n *layer) deleteContainer(ctx context.Context, idCnr cid.ID, sessionToken *session.Container) error {
```
Comment (KirillovDenis) [inline api/layer/container.go:183]: Now we can drop this function at all by moving  call `n.neoFS.DeleteContainer(ctx, idCnr, sessionToken)`  to `layer.DeleteBucket`.

**Review by KirillovDenis — APPROVED** [review]

**Pull request merged**